packages/flutter/lib/src/services/asset_manifest.dart

```diff
@@ -54,24 +58,36 @@ class _AssetManifestBin implements AssetManifest {
   }
 
   final Map<Object?, Object?> _data;
-  final Map<String, List<AssetVariant>> _typeCastedData = <String, List<AssetVariant>>{};
+  final Map<String, List<AssetMetadata>> _typeCastedData = <String, List<AssetMetadata>>{};
 
   @override
-  List<AssetVariant> getAssetVariants(String key) {
+  List<AssetMetadata> describeAssetAndVariants(String key) {
```

Comment:
imo this is a bit long of a name.

Reply:
Alternatives:

* `getAssetVariants` (original)
* `describeAssetVariants` (`describe` is slightly more precise than `get` but is perhaps unnecessary since the return type already precisely indicates what the caller is getting back.
* `getAssetAndVariants`
* `describeAssetAndVariants` (long-winded but has the least potential for confusion imo)

Any other ideas?

Reply:
getAssetVariants is fine.


Explainaing exactly what the API does is a job for a doc comment, not the name of the method.